packages/flutter_tools/lib/src/web/devfs_web.dart

```diff
@@ -44,18 +47,49 @@ class WebAssetServer {
   ///
   /// Unhandled exceptions will throw a [ToolExit] with the error and stack
   /// trace.
-  static Future<WebAssetServer> start(String hostname, int port) async {
+  static Future<WebAssetServer> start(
+    String hostname,
+    int port,
+    UrlTunneller urlTunneller,
+    BuildMode buildMode,
+    bool enableDwds,
+  ) async {
     try {
       final InternetAddress address = (await InternetAddress.lookup(hostname)).first;
       final HttpServer httpServer = await HttpServer.bind(address, port);
       final Packages packages =
           await loadPackagesFile(Uri.base.resolve('.packages'), loader: (Uri uri) => globals.fs.file(uri).readAsBytes());
-      return WebAssetServer(httpServer, packages, address,
-          onError: (dynamic error, StackTrace stackTrace) {
-        httpServer.close(force: true);
-        throwToolExit(
-            'Unhandled exception in web development server:\n$error\n$stackTrace');
-      });
+      final WebAssetServer server = WebAssetServer(httpServer, packages, address);
+
+      // In debug builds, spin up DWDS and the full asset server.
+      if (buildMode == BuildMode.debug) {
```

Comment:
nit: flip the sense and unindent the larger part.

Reply:
Done